.github/workflows/codeql-daily.yml

```diff
@@ -21,27 +26,151 @@ jobs:
           distribution: temurin
           java-version: 17
 
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
       - name: Initialize CodeQL
         uses: github/codeql-action/init@v3
         with:
-          languages: java, c-cpp
+          languages: java
+
+      - name: Build Java code
+        run: ./gradlew assemble --no-build-cache
+        # Skip build cache for full code analysis
+
+      - name: Perform CodeQL analysis
+        uses: github/codeql-action/analyze@v3
+        with:
+          category: java
+
+  # ===== C++ Analysis Job =====
+  analyze-cpp:
+    name: "Analyze C++ Code"
+    permissions:
+      actions: read
+      security-events: write
+    runs-on: windows-latest
+    
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Set up Java 17 (required for JNI compilation)
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: 17
+
+      - name: Setup Visual Studio Build Tools
+        uses: microsoft/setup-msbuild@v1
+      
+      - name: Set up Windows SDK
+        uses: ilammy/msvc-dev-cmd@v1
 
       - name: Setup Gradle
         uses: gradle/actions/setup-gradle@v4
 
-      - name: Assemble
-        # skipping build cache is needed so that all modules will be analyzed
-        run: ./gradlew assemble --no-build-cache
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: cpp
+          debug: true
+
+      - name: Build C++ code
+        shell: powershell
+        id: build-cpp
+        run: |
+          # Set required environment variables
+          $env:APPINSIGHTS_WIN10_SDK_PATH = "C:\Program Files (x86)\Windows Kits\10"
+          $env:APPINSIGHTS_VS_PATH = $env:VsInstallRoot
+          $env:JAVA_HOME = $env:JAVA_HOME_17_X64
+          
+          # Explicitly define which C++ file we're interested in
+          $sourceDir = "etw/native/src/main/cpp"
+          $headerDir = "etw/native/src/main/headers"
+          $cppFile = "$sourceDir/etw_provider.cpp"
```

Comment:
All the C++ files in this repo exist in this directory:
https://github.com/microsoft/ApplicationInsights-Java/tree/main/etw/native/src/main